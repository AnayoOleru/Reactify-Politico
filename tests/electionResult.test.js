import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ElectionResultStore, { Results } from '../src/components/pages/election-results.jsx';
import UserNavbar from '../src/components/reuseable component/user-navbar.component.jsx';


const props = {
  posts: {},
  getAllElectionResults: jest.fn(() => Promise.resolve()),
  candidates: [{
    candidate:'',
    result:'',
}],
match: {
  params: {
    officeId:1
  }
}
};

const initialState = {
  get: {
      success: true,
      candidates: {
          title: '',
          content: ''
      },
      error: ''
  },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

const instance = () => mount(<ElectionResultStore />);

const OfficeComponent = (
  <Provider store={store}>
    <Results {...props} />
  </Provider>
);

describe('Alloffice component', () => {
     it('render component', () => {
      const wrapper = shallow(<Results {...props} />);
      expect(wrapper).toMatchSnapshot();
     });
});

describe('Office component', () => {
  it('should match snapshot', () => {
    const wrap = OfficeComponent;
    expect(wrap).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(OfficeComponent);
    expect(wrapper.find(<UserNavbar {...props} />));
  });

  it('has map state to props', () => {
    const component = shallow(<ElectionResultStore {...props} store={store} />);
    expect(component.length).toBe(1);
  });

});
