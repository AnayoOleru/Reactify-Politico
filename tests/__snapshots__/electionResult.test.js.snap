// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alloffice component render component 1`] = `ShallowWrapper {}`;

exports[`Office component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Results
    candidates={
      Array [
        Object {
          "candidate": "",
          "result": "",
        },
      ]
    }
    getAllElectionResults={[MockFunction]}
    match={
      Object {
        "params": Object {
          "officeId": 1,
        },
      }
    }
    posts={Object {}}
  />
</Provider>
`;
